src/ir/child-typer.h

```diff
@@ -81,6 +81,20 @@ template<typename Subtype> struct ChildTyper : OverriddenVisitor<Subtype> {
     note(ptrp, wasm.getMemory(mem)->addressType);
   }
 
+  Table* getTable(Name table) {
+    if (table) {
+      return wasm.getTable(table);
+    }
+
+    // No table specified, so this is the default.
+    assert(!wasm.tables.empty());
+    return wasm.tables[0].get();
+  }
```

Comment:
@tlively why is this necessary for tables but not memories? That is, when parsing some `table.get` instructions there is no name for the table. That is the case for e.g.
```wat
 (func $table-get
  i32.const 0
  table.get
  drop
)
```
where there is no table specified. But this does not seem to be the case for memories, where the memory name is always non-null? (see `notePointer`, above, which does not need to be careful like `noteTableIndex`)

Reply:
Oops, just saw this question. Will take a look.

Reply:
Hmm, so when the table or memory index is left implicit in the parsed text, `ParseDefsCtx::getTable` and `ParseDefsCtx::getMemory` should be responsible for looking up the internal name of table or memory 0. So the table name should always exist here, and if it doesn't, the bug is earlier in the parser.